recipes-bsp/u-boot/files/0006-dts-iot2050-Support-new-IOT2050-SM-variant.patch

```diff
@@ -0,0 +1,634 @@
+ {
+ 	ulong m2_manual_config = env_get_ulong("m2_manual_config", 10,
+@@ -288,10 +315,6 @@ static void m2_connector_setup(void)
+ 	struct m2_config_pins config_pins;
+ 	unsigned int n;
+ 
+-	/* enable M.2 connector power */
+-	set_pinvalue("gpio@601000_17", "P3V3_M2_EN", 1);
+-	udelay(4 * 100);
+-
+ 	if (m2_manual_config < CONNECTOR_MODE_INVALID) {
+ 		mode_info = " [manual mode]";
+ 		connector_mode = m2_manual_config;
+@@ -339,25 +362,50 @@ int board_init(void)
+ 
+ int dram_init(void)
+ {
+-	if (board_is_advanced())
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++#ifdef CONFIG_PHYS_64BIT
++	gd->ram_size = ((phys_size_t)(info->ddr_size_mb)) << 20;
++#else
++	if (info->ddr_size_mb > 2048)
+ 		gd->ram_size = SZ_2G;
+ 	else
+-		gd->ram_size = SZ_1G;
+-
++		gd->ram_size = ((phys_size_t)(info->ddr_size_mb)) << 20;
++#endif
+ 	return 0;
+ }
+ 
++ulong board_get_usable_ram_top(ulong total_size)
++{
++#ifdef CONFIG_PHYS_64BIT
++	/* Limit RAM used by U-Boot to the DDR low region */
++	if (gd->ram_top > 0x100000000)
++		return 0x100000000;
++#endif
++
++	return gd->ram_top;
++}
++
+ int dram_init_banksize(void)
+ {
+ 	dram_init();
+ 
+-	/* Bank 0 declares the memory available in the DDR low region */
+-	gd->bd->bi_dram[0].start = CFG_SYS_SDRAM_BASE;
+-	gd->bd->bi_dram[0].size = gd->ram_size;
++	if (gd->ram_size > SZ_2G) {
++		/* Bank 0 declares the memory available in the DDR low region */
++		gd->bd->bi_dram[0].start = CFG_SYS_SDRAM_BASE;
++		gd->bd->bi_dram[0].size = SZ_2G;
++
++		/* Bank 1 declares the memory available in the DDR high region */
++		gd->bd->bi_dram[1].start = CFG_SYS_SDRAM_BASE1;
++		gd->bd->bi_dram[1].size = gd->ram_size - SZ_2G;
++	} else {
++		/* Bank 0 declares the memory available in the DDR low region */
++		gd->bd->bi_dram[0].start = CFG_SYS_SDRAM_BASE;
++		gd->bd->bi_dram[0].size = gd->ram_size;
+ 
+-	/* Bank 1 declares the memory available in the DDR high region */
+-	gd->bd->bi_dram[1].start = 0;
+-	gd->bd->bi_dram[1].size = 0;
++		/* Bank 1 declares the memory available in the DDR high region */
++		gd->bd->bi_dram[1].start = 0;
++		gd->bd->bi_dram[1].size = 0;
++	}
+ 
+ 	return 0;
+ }
+@@ -429,6 +477,8 @@ int board_late_init(void)
+ 	/* change CTRL_MMR register to let serdes0 not output USB3.0 signals. */
+ 	writel(0x3, SERDES0_LANE_SELECT);
+ 
++	enable_pcie_connector_power();
++
+ 	if (board_is_m2())
+ 		m2_connector_setup();
+ 
+@@ -443,7 +493,7 @@ int board_late_init(void)
+ }
+ 
+ #if defined(CONFIG_OF_LIBFDT) && defined(CONFIG_OF_BOARD_SETUP)
+-static void m2_fdt_fixup(void *blob)
++static void variants_fdt_fixup(void *blob)
+ {
+ 	void *overlay_copy = NULL;
+ 	void *fdt_copy = NULL;
+@@ -483,14 +533,14 @@ cleanup:
+ 	return;
+ 
+ fixup_error:
+-	pr_err("Could not apply M.2 device tree overlay\n");
++	pr_err("Could not apply device tree overlay\n");
+ 	goto cleanup;
+ }
+ 
+ int ft_board_setup(void *blob, struct bd_info *bd)
+ {
+ 	if (board_is_m2())
+-		m2_fdt_fixup(blob);
++		variants_fdt_fixup(blob);
```

Comment:
Maybe not a good time to do it anyway: I was thinking of promoting this part of logic to be more common... not only for m.2... just in case we have more overlays variants - what we'd done for SM but dropped just days before.

Reply:
Or maybe you are wondering why this fixup is not applied to SM? - We've dropped all the overlays for SM due to:
1. USB3 issue
2. Some card uses the PERST pin even it's not using PCIE signal (4g card using usb2 signal), so a no-high-speed overlay is also not usable for such cards.

However I think this fdt fixup logic should be common - just in case we have another variant in the future that need overlay fixup as well.

Reply:
As above: The rework bits with reasoning should become separate.

Reply:
Separated.